crates/bitcoin-da/src/spec/header.rs

```diff
@@ -41,28 +47,113 @@ impl BlockHeaderTrait for HeaderWrapper {
 
 impl HeaderWrapper {
     pub fn new(
-        header: Header,
+        header: BitHeader,
         tx_count: u32,
         height: u64,
         txs_commitment: WitnessMerkleNode,
     ) -> Self {
         Self {
-            header,
+            header: header.into(),
             tx_count,
             height,
-            txs_commitment,
+            txs_commitment: txs_commitment.to_byte_array(),
         }
     }
 
     pub fn block_hash(&self) -> BlockHash {
-        self.header.block_hash()
+        let bit_header = BitHeader::from(&self.header);
+        bit_header.block_hash()
     }
 
-    pub fn merkle_root(&self) -> TxMerkleNode {
+    pub fn merkle_root(&self) -> [u8; 32] {
         self.header.merkle_root
     }
+}
+
+#[derive(
+    Copy,
+    Debug,
+    PartialEq,
+    Eq,
+    Clone,
+    PartialOrd,
+    Ord,
+    Hash,
+    BorshDeserialize,
+    BorshSerialize,
+    Serialize,
+    Deserialize,
+)]
+pub struct OurHeader {
```

Comment:
Commented on the name above. The `Our` prefix is not needed.

Reply:
this one's a bit tricky, what should be the name?

BitcoinHeaderWrapper?